ios/RNSScreenStack.mm

```diff
@@ -57,6 +57,31 @@ - (UIViewController *)childViewControllerForStatusBarHidden
   return [self topViewController];
 }
 
+- (void)viewDidLayoutSubviews
+{
+  if ([self.topViewController isKindOfClass:[RNSScreen class]]) {
+    RNSScreen *screenController = (RNSScreen *)self.topViewController;
+
+    // If RNSScreen includes a navigation controller of type RNScreensNavigationController, it should not calculate
+    // header height, as it could have nested stack.
```

Comment:
What about both navigators having headers? I know it is a weird case but should we care about it?

Reply:
If you mean that we could have a nested stack with more than one header (which is a typical situation for nested stacks) then only the last one should call `calculateHeaderHeightIsModal`, so I assume we're right now caring about it (If I understand you correctly 😄)